README.md
# Indiv Theme

[Indiv](https://indiv.no) theme, made for [Theme UI](https://theme-ui.com).

## Quickstart

```shell
▶ yarn add @indivorg/theme
```

## Development

```shell
▶ yarn dev
```

The command above starts a Next.js instance. We've laid out all components from
Theme UI, so you can easily stylize these with the theme (with hot reload etc.).


CHANGELOG.md
# Changelog

### [4.0.1](https://www.github.com/indivorg/theme/compare/v4.0.0...v4.0.1) (2023-05-14)


### Bug Fixes

* horizontal line should only be 1px in height ([#55](https://www.github.com/indivorg/theme/issues/55)) ([4bb1ecd](https://www.github.com/indivorg/theme/commit/4bb1ecd06695768f1f44c46d12d73af6b3f098a6))

## [4.0.0](https://www.github.com/indivorg/theme/compare/v3.7.0...v4.0.0) (2021-11-30)


### ⚠ BREAKING CHANGES

* Remove unstyled button (#53)

### Code Refactoring

* Remove unstyled button ([#53](https://www.github.com/indivorg/theme/issues/53)) ([d3cb750](https://www.github.com/indivorg/theme/commit/d3cb7506605042891b6db5b8ce409a1b00b74518))

## [3.7.0](https://www.github.com/indivorg/theme/compare/v3.6.0...v3.7.0) (2021-11-25)


### Features

* add dividers to theme samples ([713b7ee](https://www.github.com/indivorg/theme/commit/713b7ee47e01059a529d956eef696009bc374db4))
* implement new badge style ([5279940](https://www.github.com/indivorg/theme/commit/5279940e2c5207c94fd41e22e9041ef43fb2765b))
* implement new divider style ([9ede1b6](https://www.github.com/indivorg/theme/commit/9ede1b67262181605a262baf8674e591f1f95461))


### Bug Fixes

* remove px values ([f31dec8](https://www.github.com/indivorg/theme/commit/f31dec88c72fc5100fece5a1e4321c7eecd21439))
* set borderStyle instead of border ([4f66e5e](https://www.github.com/indivorg/theme/commit/4f66e5e316028e4bb1860f7fa2cf685617081ff8))
* typo ([5beae5e](https://www.github.com/indivorg/theme/commit/5beae5e7c3e96ce8ebec9c998f16be914c78e14e))
* update with new design ([cd686bc](https://www.github.com/indivorg/theme/commit/cd686bc8094054d865364d2eee6c4bca6ff2d298))

## [3.6.0](https://www.github.com/indivorg/theme/compare/v3.5.0...v3.6.0) (2021-11-24)


### Features

* implement new heading styles ([e26f5ad](https://www.github.com/indivorg/theme/commit/e26f5ad0c64e834c78ccf12694f89868048b4541))
* rework unstyled variant to not look like button ([611e87b](https://www.github.com/indivorg/theme/commit/611e87bdb42f5a098ec0cd379a66123898bbf2e4))

## [3.5.0](https://www.github.com/indivorg/theme/compare/v3.4.0...v3.5.0) (2021-10-26)


### Features

* Improve button styles ([#38](https://www.github.com/indivorg/theme/issues/38)) ([6c71fb1](https://www.github.com/indivorg/theme/commit/6c71fb15b3bf52eed05ae7e5f7ec0e1e514293e5))

## [3.4.0](https://www.github.com/indivorg/theme/compare/v3.3.0...v3.4.0) (2021-10-24)


### Features

* Add margins on paragraphs ([#33](https://www.github.com/indivorg/theme/issues/33)) ([b714dd2](https://www.github.com/indivorg/theme/commit/b714dd2fb555e0c7db5fa508d305207fb2eefdca))
* disable buttons style ([#36](https://www.github.com/indivorg/theme/issues/36)) ([4d3a0d8](https://www.github.com/indivorg/theme/commit/4d3a0d8dbf9ed5730cd747962aef17411c4dde40))


### Bug Fixes

* remove px ([#35](https://www.github.com/indivorg/theme/issues/35)) ([b41c2d0](https://www.github.com/indivorg/theme/commit/b41c2d0dc7c48bec47b6a601b050ade140211ece))

## [3.3.0](https://www.github.com/indivorg/theme/compare/v3.2.0...v3.3.0) (2021-10-18)


### Features

* Add readable layout and misc. ([#30](https://www.github.com/indivorg/theme/issues/30)) ([bec020f](https://www.github.com/indivorg/theme/commit/bec020f31253b6fddfc76a54ef2208af8ab42926))

## [3.2.0](https://www.github.com/indivorg/theme/compare/v3.1.1...v3.2.0) (2021-10-13)


### Features

* add unstyled button variant ([60a0f8c](https://www.github.com/indivorg/theme/commit/60a0f8cfc4b5836a5d0c34e3ba2a00d34d5a2135))

### [3.1.1](https://www.github.com/indivorg/theme/compare/v3.1.0...v3.1.1) (2021-10-01)


### Bug Fixes

* main file reference ([b033b68](https://www.github.com/indivorg/theme/commit/b033b6876e505152f70c43dd241e48573e6f466e))

## [3.1.0](https://www.github.com/indivorg/theme/compare/v3.0.0...v3.1.0) (2021-09-30)


### Features

* Add antialiasing ([#25](https://www.github.com/indivorg/theme/issues/25)) ([9a2fa52](https://www.github.com/indivorg/theme/commit/9a2fa52d1ce900dbcadc02a476e6921d560d6bc8))


### Bug Fixes

* button text colors ([f1056b1](https://www.github.com/indivorg/theme/commit/f1056b1d8ea58effd32800a6467569a5bec841c7))
* preview issues ([8d424d8](https://www.github.com/indivorg/theme/commit/8d424d82ef3d581bf62e077cbe170f3eb544c2b8))

## [3.0.0](https://www.github.com/indivorg/theme/compare/v2.0.0...v3.0.0) (2021-09-29)


### ⚠ BREAKING CHANGES

* Improve buttons (#22)
* Add webTheme (#18)
* Better button styles and development environment for styling (#15)

### Features

* Add webTheme ([#18](https://www.github.com/indivorg/theme/issues/18)) ([478a034](https://www.github.com/indivorg/theme/commit/478a034d109653043aa6e1cf8d0486a366457677))
* Better button styles and development environment for styling ([#15](https://www.github.com/indivorg/theme/issues/15)) ([99b6679](https://www.github.com/indivorg/theme/commit/99b6679a2ae6bf35c74ef49393430c46e0a77eb1))
* Improve buttons ([#22](https://www.github.com/indivorg/theme/issues/22)) ([4e48d44](https://www.github.com/indivorg/theme/commit/4e48d44c2b686aa5e55fee4c463a37b114389227))
* re-add removed code ([8e896dc](https://www.github.com/indivorg/theme/commit/8e896dcc0087b5cb37d26776a60f9a9ac6b114c1))
* update theme with latest theme from indivorg/app ([b0ca63c](https://www.github.com/indivorg/theme/commit/b0ca63c3089abf84ca33b1c1f4293d341071f49d))


### Bug Fixes

* add correct text color ([e35b020](https://www.github.com/indivorg/theme/commit/e35b0202e017a5d10d7a93b0e62cbb7d7fa97b49))
* add linebreak ([a33c611](https://www.github.com/indivorg/theme/commit/a33c611c756e632a08a6c31e7ad922a799a2509c))

## [2.0.0](https://www.github.com/indivorg/theme/compare/v1.0.2...v2.0.0) (2021-07-18)

### ⚠ BREAKING CHANGES

- change to `@indivorg` organization

### Features

- change to `[@indivorg](https://www.github.com/indivorg)` organization
  ([9260229](https://www.github.com/indivorg/theme/commit/92602290db1f675351918bd874d06e9cdb3a4ad1))
- move from `indivno` to `indivorg`

src/theme.ts
import type { Theme, ThemeUIStyleObject } from 'theme-ui';
import { alerts } from './alerts';
import { buttons } from './button';
import { colors } from './colors';
import { layout } from './layout';
import { links } from './links';
import { styles } from './styles';

export interface IndivTheme extends Theme {
  label: ThemeUIStyleObject;
}

export const theme: IndivTheme = {
  sizes: {
    container: 1205,
    readable: 770,
  },
  breakpoints: ['40em', '52em', '64em'],
  space: [0, 4, 8, 16, 32, 64, 128, 256, 512],
  radii: [0, 2, 4, 6, 8, 16, 32],
  borderWidths: [0, 2],
  fonts: {
    body: 'system-ui, -apple-system, "Public Sans", sans-serif',
    heading: 'inherit',
    monospace: 'Menlo, monospace',
  },
  cards: {
    primary: {
      color: 'text',
      p: [3, 4],
      borderRadius: 'normal',
      overflow: 'hidden',
    },
    secondary: {
      bg: 'muted',
      borderRadius: 3,
      p: 4,
    },
  },
  fontSizes: [12, 14, 16, 18, 20, 24, 32, 48, 64, 74],
  fontWeights: {
    light: 300,
    body: 400,
    heading: 700,
    bold: 700,
  },
  lineHeights: {
    body: 1.5,
    heading: 1.125,
  },
  text: {
    headline: {
      lineHeight: 'heading',
      fontWeight: 'heading',
      fontSize: 9,
      py: 3,
    },
    heading: {
      lineHeight: 'heading',
      fontWeight: 'light',
      fontSize: 7,
      py: 3,
    },
    smallHeadline: {
      lineHeight: 'heading',
      fontWeight: 'heading',
      fontSize: 6,
      py: 2,
    },
    tinyHeading: {
      lineHeight: 'heading',
      fontWeight: 'light',
      fontSize: 4,
      py: 2,
    },
    tinyHeadline: {
      lineHeight: 'heading',
      fontWeight: 'heading',
      fontSize: 3,
      py: 2,
    },

    paragraph: {
      my: 2,
    },
    block: {
      variant: 'paragraph',
      textAlign: 'justify',
      textAlignLast: 'start',
      textJustify: 'auto',
    },
  },
  forms: {
    plain: {
      border: 0,
      padding: 0,
      outline: 'none',
    },
  },
  label: {
    fontWeight: 'bold',
    color: 'text',
    fontSize: 1,
  },
  badges: {
    primary: {
      border: '3px solid',
      borderColor: 'secondary',
      borderRadius: 4,
      py: 1,
      px: 2,
      bg: 'lightGreen',
      color: 'text',
      fontWeight: 'body',
    },
    secondary: {
      border: '3px solid',
      borderColor: 'secondary',
      borderRadius: 4,
      py: 1,
      px: 2,
      bg: 'background',
      color: 'primary',
      fontWeight: 'body',
    },
  },
  images: {
    rounded: {
      borderRadius: '50%',
    },
  },
  colors,
  styles,
  alerts,
  layout,
  buttons,
  links,
};
